BridgeAppSDK/SBAScheduledActivityManager.swift

```diff
@@ -140,27 +141,36 @@ public class SBAScheduledActivityManager: NSObject, SBASharedInfoController, ORK
     
     public func filterPredicateForScheduledActivitySection(section: SBAScheduledActivitySection) -> NSPredicate? {
         
-        let expiredKey = "expiredOn"
+        // syoung 05/06/2016 As of Swift 2.2, there is no equivilent to NSStringFromSelector
+        let expiredKey = "expiresOn"
         let scheduledKey = "scheduledOn"
         let finishedKey = "finishedOn"
+        let optionalKey = "persistent"
+        
+        // For unscheduled we need to also filter out those tasks that were finished prior to today
+        let finishedToday = NSPredicate(day: NSDate(), dateKey: finishedKey)
+        let unfinished = NSPredicate(format: "%K == nil", finishedKey)
+        let optionalFilter = NSPredicate(format: "%K == 1", optionalKey)
+        let unfinishedOrFinishedToday = NSCompoundPredicate(orPredicateWithSubpredicates: [unfinished, finishedToday])
+        let expiredYesterday = NSPredicate(day: NSDate().dateByAddingTimeInterval(-24*60*60), dateKey: expiredKey)
+        let notExpired = NSCompoundPredicate(notPredicateWithSubpredicate: expiredYesterday)
+        let notOptional = NSCompoundPredicate(notPredicateWithSubpredicate: optionalFilter)
+        let calendar = NSCalendar.currentCalendar()
+        let tomorrow = calendar.startOfDayForDate(NSDate().dateByAddingTimeInterval(24*60*60))
+        let notFuture = NSPredicate(format: "%K == nil OR %K < %@", scheduledKey, scheduledKey, tomorrow)
         
         switch section {
             
-        case .Yesterday:
-            // only expired schedules for yesterday
-            return NSPredicate(day: NSDate().dateByAddingTimeInterval(-24*60*60), dateKey: expiredKey)
+        case .ExpiredYesterday:
+            // expired yesterday section only showns those expired tasks that are also unfinished
+            return NSCompoundPredicate(andPredicateWithSubpredicates: [unfinished, expiredYesterday])
             
         case .Today:
-            // scheduled for today only
-            return NSPredicate(day: NSDate(), dateKey: scheduledKey)
+            return NSCompoundPredicate(andPredicateWithSubpredicates: [notOptional, notFuture, unfinishedOrFinishedToday, notExpired])
```

Comment:
Yes, but bridge only caches back one day and I was in a hurry. ;)

Sent from my iPhone

> On May 6, 2016, at 2:34 PM, Erin-Mounts <notifications@github.com> wrote:
> 
> In BridgeAppSDK/SBAScheduledActivityManager.swift:
> 
> >              
> >          case .Today:
> > -            // scheduled for today only
> > -            return NSPredicate(day: NSDate(), dateKey: scheduledKey)
> > +            return NSCompoundPredicate(andPredicateWithSubpredicates: [notOptional, notFuture, unfinishedOrFinishedToday, notExpired])
> In Lilly we're filtering to only one day in the past, but other apps e.g. FPHS may keep a longer history -- wouldn't this end up showing things in Today section that expired a few days ago?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>